pennylane/transforms/diagonalize_measurements.py

```diff
@@ -60,6 +60,14 @@ def diagonalize_measurements(tape, supported_base_obs=_default_supported_obs, to
         circuit measurements, the :func:`split_non_commuting <pennylane.transforms.split_non_commuting>` transform
         can be applied.
 
+    .. note::
+        This transform will diagonalize what it can, i.e., ``qml.X``, ``qml.Y``, ``qml.Z``,
+        ``qml.Hadamard``, ``qml.Identity``, or a linear combination of them. Any unrecognized
+        observable will not raise an error, deferring to the device's validation for supported
+        measurements later on. Lastly, if ``diagonalize_measurements`` produces additional gates
+        that the device does not support, the :func:`~pennylane.devices.preprocess.decompose``
+        transform should be applied to ensure that the additional gates are decomposed to those
+        that the device supports.
```

Comment:
@astralcai re: above, how about this?

```suggestion
        transform is applied by PennyLane devices by default to ensure that the additional gates are decomposed to those
        that the device supports. 
```

Reply:
I don't think that's true, because most PennyLane devices such as `default.qubit` don't actually apply the `diagonalize_measurements` transform, because they already support measuring all of `X`, `Y`, `Z`, and `Hadamard`. This note is meant for device developers thinking about adding `diagonalize_measurements` to their own `preprocess` implementation, to remind them that if you choose to include this transform, make sure that your `decompose` is placed after it, not before it. This is automatically the case if the device developer does not override the `preprocess` but simply use the default implementation inherited from the device API, but if they are reading the docs for this transform, they are most likely implementing their own `preprocess`, which means that nothing is done by default for them and they need to remember to do this.

Reply:
> if they are reading the docs for this transform, they are most likely implementing their own preprocess, which means that nothing is done by default for them and they need to remember to do this.

I don't think we should assume this! I imagine a user being confused as to why their circuit has gates in it that they didn't put there, so they go down the rabbit hole which leads to `diagonalize_measurements`. In this docstring, we should be clear about when `decompose` is applied and, secondarily, when it _should_ be applied (if that makes sense). 

Having said all of this, I've marinated over your feedback and I think it's probably more confusing to try and elaborate on this point than to leave what's currently there. So, I'm fine with the docstring as written 👍.

Reply:
Feel free to resolve this thread and go ahead with the PR 🙂. We can always come back to this if it turns out that a user gets confused